detectable/src/main/java/com/blackduck/integration/detectable/detectables/dart/pubspec/PubSpecLockParser.java

```diff
@@ -39,11 +39,6 @@ public DependencyGraph parse(List<String> pubSpecLockLines) {
                 dependencyName = parseValueFromLine(trimmedLine, NAME_LINE_KEY);
             } else if (inPackages && trimmedLine.startsWith(VERSION_LINE_KEY)) {
                 Optional<String> dependencyVersion = parseValueFromLine(trimmedLine, VERSION_LINE_KEY);
-                if (dependencyName.isPresent() && dependencyVersion.isPresent()) {
-                    dependencyGraph.addChildToRoot(DependencyCreator.nameVersion(Forge.DART, dependencyName.get(), dependencyVersion.get()));
-                    // After process dependency, reset name and version variables
-                    dependencyName = Optional.empty();
-                }
```

Comment:
I had encountered this when releasing 10.0.0 but I didn't want to remove the code as it may happen that dependencyName is being assigned during execution of for loop. I do not think we should entirely remove the code but test this around.

Reply:
I dont see where dependencyName could get assigned within this for loop. Maybe I'm missing it, could you please point it out?

Reply:
On line 39, lets say we are inDescription section as per loop, then dependencyName can be assigned there.

Reply:
If line 39 was executed, line 40 (lines 42 through 46 are nested in here also) would never execute because of the "else if", no?

Reply:
Yes correct but they could be executed in the next iteration.

Reply:
> It's interesting that the file is parsed line-by-line. I looked at some `pubspec.lock` examples and it looks like it's Yaml. This code expects a specific order of keys in a yaml object.
> 
> My vote would be to leave the code as-is for this release but consider creating a refactoring ticket to parse the file as Yaml. This would make parsing more reliable, especially if the order of keys ever changes.

I second this

Reply:
Created IDETECT-4581

Reply:
Changes have been reverted leaving PubSpecLockParser as is